lib/spack/spack/cmd/buildcache.py

```diff
@@ -495,6 +498,15 @@ def install_tarball(spec, args):
     else:
         tarball = bindist.download_tarball(spec)
         if tarball:
+            if args.sha256:
+                checker = spack.util.crypto.Checker(args.sha256)
+                msg = ('cannot verify checksum for "{0}"'
+                       ' [expected={1}]')
+                msg = msg.format(tarball, args.sha256)
+                if not checker.check(tarball):
+                    raise RuntimeError(msg)
```

Comment:
This should probably be a `spack.binary_distribution.NoChecksumException`

Reply:
Fixed in 50ea6e5